geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -780,6 +795,9 @@ public void waitForMessageState(ID sender, Map<String, Long> state)
         JGAddress to = new JGAddress(mbr);
         short version = mbr.getVersionOrdinal();
         org.jgroups.Message jmsg = messages.get(version);
+        if (jmsg == null) {
```

Comment:
this and `failedRecepients` look to be fixing an existing bug.... are there new test(s) for these changes?

Reply:
there are existing tests for failedRecipients.  There are many places that don't check for failedRecipients but they are covered by tests that shutdown/kill other members.

Reply:
I am mainly concerned that functional behavior was being changed with this refactoring.